Khan/wonder-blocks
# Add new SearchField component

## Summary:
Added new SearchField component
- search-field.js
- search-field.test.js
- search-field.stories.js

Issue: https://khanacademy.atlassian.net/browse/WB-1117

Demo:
https://user-images.githubusercontent.com/13231763/160504207-b0ce9594-21c1-4de1-a7ce-460320e35de2.mov


Docs:
<img width="1747" alt="Screen Shot 2022-03-28 at 4 30 35 PM" src="https://user-images.githubusercontent.com/13231763/160503895-4fba3d2e-27cd-4c01-aba1-d1b1a69203c3.png">

Light and Disabled stories:
<img width="1081" alt="image" src="https://user-images.githubusercontent.com/13231763/160505100-c6b65772-bc5f-4d19-b4f9-096adaf9eb24.png">

Light mode focused:
<img width="996" alt="Screen Shot 2022-03-28 at 4 32 24 PM" src="https://user-images.githubusercontent.com/13231763/160504117-7f8666de-84ac-445e-abc6-c57d3e9389a6.png">

Light mode focused with X:
<img width="1010" alt="Screen Shot 2022-03-28 at 4 33 02 PM" src="https://user-images.githubusercontent.com/13231763/160504144-a1df5fcd-effa-4112-9bbb-ec285e402804.png">



## Test plan:
`yarn test packages/wonder-blocks-search-field/src/components/__tests__/search-field.test.js`

Manual testing in storybook
- Check that you can type in the textbox
- Clear button should show up after typing
- Clear button should go away if the field is clear
- Check spacing and borders
- Test focusing in and out with clicking and with keyboard nav
